pytorch_lightning/loops/fit_loop.py

```diff
@@ -0,0 +1,254 @@
+# limitations under the License.
+
+import logging
+from contextlib import suppress
+from typing import List, Optional, Tuple
+
+from torch.optim import Optimizer
+
+import pytorch_lightning as pl
+from pytorch_lightning.loops.base import Loop
+from pytorch_lightning.loops.training_epoch_loop import TrainingEpochLoop
+from pytorch_lightning.trainer.connectors.logger_connector.result import ResultCollection
+from pytorch_lightning.trainer.supporters import TensorRunningAccum
+from pytorch_lightning.utilities import rank_zero_info
+
+log = logging.getLogger(__name__)
+
+
+class FitLoop(Loop):
+
+    def __init__(
+        self,
+        min_epochs: Optional[int] = None,
+        max_epochs: Optional[int] = None,
+        min_steps: Optional[int] = None,
+        max_steps: Optional[int] = None
+    ):
+        super().__init__()
+        # If neither max_epochs or max_steps is set, then use existing default of max_epochs = 1000
+        self.max_epochs = 1000 if (max_epochs is None and max_steps is None) else max_epochs
+        # If neither min_epochs or min_steps is set, then use existing default of min_epochs = 1
+        self.min_epochs = 1 if (min_epochs is None and min_steps is None) else min_epochs
+
+        self.training_loop = TrainingEpochLoop(min_steps, max_steps)
+        self.results = ResultCollection(training=True)
+
+    @property
+    def current_epoch(self) -> int:
+        return self.iteration_count
+
+    @current_epoch.setter
+    def current_epoch(self, value: int):
+        self.iteration_count = value
+
+    @property
+    def global_step(self):
+        return self.training_loop.global_step
+
+    @global_step.setter
+    def global_step(self, value):
+        self.training_loop.global_step = value
+
+    @property
+    def total_batch_idx(self):
+        return self.training_loop.total_batch_idx
+
+    @property
+    def batch_idx(self):
+        return self.training_loop.iteration_count
+
+    @property
+    def split_idx(self):
+        return self.training_loop.split_idx
+
+    @property
+    def min_steps(self):
+        return self.training_loop.min_steps
+
+    @property
+    def max_steps(self):
+        return self.training_loop.max_steps
+
+    @max_steps.setter
+    def max_steps(self, value):
+        # TODO(@awaelchli): This setter is required by debugging connector (fast dev run), should be avoided
+        self.training_loop.max_steps = value
+
+    @property
+    def running_loss(self):
+        return self.training_loop.batch_loop.running_loss
+
+    @property
+    def skip_backward(self) -> bool:
+        """ Determines whether the loop will skip backward during automatic optimization. """
+        return self.training_loop.batch_loop.skip_backward
+
+    @skip_backward.setter
+    def skip_backward(self, value: bool):
+        """ Determines whether the loop will skip backward during automatic optimization. """
+        self.training_loop.batch_loop.skip_backward = value
+
```

Comment:
why elevate this to a public API? before, the callback was accessing a private property

Reply:
so you are saying this is ok? but protected access is only supposed to be happening within the same class hierarchy, i.e., the class or the subclass. I made it a public precisely because I saw that a callback from outside is accessing it as a protected member.

Reply:
yes, i dont think this should be explicitly exposed to end users. just because callbacks from outside are reaching in doesn't mean we should expose it

Reply:
ok. reverted back to `_skip_backard` as before!